src/realm/group_shared.cpp

```diff
@@ -1809,9 +1840,27 @@ void SharedGroup::do_end_read() noexcept
 }
 
 
+
+bool SharedGroup::do_try_begin_write()
+{
+    // In the non-blocking case, we will only succeed if there is no contention for
+    // the write mutex. For this case we are trivially fair and can ignore the
+    // fairness machinery.
+    bool got_the_lock = m_writemutex.try_lock();
+    if (!got_the_lock) {
+        return false;
+    }
+    else {
+        finish_begin_write();
+        return true;
+    }
```

Comment:
This is not really a biggie, but as I read it, the `if (!got_the_lock)` is an early out and thus, the `else` can be removed for clarity. If it is deemed important to keep both "branches" as is, consider swapping them to avoid the negation.